### PR TITLE
Add Ruby implementation for Day 6 problem; Add Ruby, C++ & Java implementations for Day 7 problem 

### PR DESCRIPTION
**Pull request by aaditkamat**
Please make sure you have done the following:

- [x] Created a folder with the name of language you code inside `/Day<n>/` directory (Where n is for nth day)
- [x] Added your Codes
- [x] Tested your codes
- [x] Added credits (author and date) at the top of the code
- [x] Updated README.md
- [x] Added yourself as a contributor using `all-conributors-cli`

### CODE REVIEW

**Comment by aaditkamat** [comment]
Sorry for these extraneous change in titles, I'm trying to format the title properly

File: day6/README.md [hunk 1]
```diff
@@ -223,74 +223,32 @@ def capitalize_sentence_long(string):
 
 print("Enter a string: ", end= ' ')
 string = input()
-print("String \'", string, "\' with first letter of each word capitalized (short version): ", capitalize_sentence_short(string))
-print("String \'", string, "\' with first letter of each word capitalized (slightly long version): ", capitalize_sentence_long(string))
+print("String \'", string, "\' with first letter of each word capitalized (short version): ", capitalize_sentence_short(string), sep='')
+print("String \'", string, "\' with first letter of each word capitalized (slightly long version): ", capitalize_sentence_long(string), sep='')
 ```
 
-## C++ Implementation
```
Comment (MadhavBahlMD) [inline day6/README.md]: Why have the C++ Implementations been removed?

File: day6/README.md [hunk 2]
```diff
@@ -223,74 +223,32 @@ def capitalize_sentence_long(string):
...
 
-### [Solution](./C++/day6_1.cpp)
+### [sentence_capitalization.rb](./Ruby/sentence_capitalization.rb)
 
-```cpp
-/*
-* @author : dhruv-gupta14
-* @date : 27/12/2018
-*/
+```ruby
```
Comment (MadhavBahlMD) [inline day6/README.md]: Where are the `@author` and `@date` credits

File: day6/README.md [hunk 3]
```diff
@@ -487,77 +445,32 @@ def reverse_words(string):
 
 print("Enter a string: ", end= '')
 string = input()
-print("Reverse of string: ", reverse_words(string)})
-```
-
-## C++ Implementation
```
Comment (MadhavBahlMD) [inline day6/README.md]: Why did you remove the C++ implementation

File: day6/README.md [hunk 4]
```diff
@@ -750,90 +663,41 @@ second_str = input()
...
+    first_word_dict == second_word_dict
+end
+
+puts "Enter two strings: "
+first_str = gets().chomp
+second_str = gets().chomp
+puts "\nAre #{first_str} and #{second_str} anagrams? #{check_anagram(String.new(first_str), String.new(second_str))}"
 ```
-
-### [Anagram Check by @imkaka](./C++/checkAnagram.cpp)
```
Comment (MadhavBahlMD) [inline day6/README.md]: Why was this removed?

File: day6/README.md [hunk 5]
```diff
@@ -487,77 +445,32 @@ def reverse_words(string):
...
-    {
-        reverse(ans.begin(),ans.end());
-        cout << ans << " ";
-    }
-    return 0;
-}
+print("Reverse of string \'", string, "\': ", reverse_words(string), sep='')
 ```
 
-### [Word Reverse by @imkaka](./C++/wordReverse.cpp)
```
Comment (MadhavBahlMD) [inline day6/README.md]: Why was this removed?
